### ISSUE TITLE
fcos installer doubles 99_openshift-machineconfig_99-master-ssh.yaml in bootstrap

### ISSUE THREAD

**Issue by gtema** [openshift/okd#63]
# Version

```console
$ openshift-install version
openshift-install unreleased-master-2411-g1dce0a90471670f87843c7824764edade9cfcb12
built from commit 1dce0a90471670f87843c7824764edade9cfcb12
release image registry.svc.ci.openshift.org/origin/release:4.3
```

# Platform:

baremetal/UPI

# What happened?
`openshift create ignition-configs` duplicates 99_openshift-machineconfig_99-master-ssh.yaml and 99_openshift-machineconfig_99-worker-ssh.yaml in the bootstrap.ign, what results in machines refusing this ignition

```
gtema@localhost:test$  openshift-install create manifests
INFO Consuming Install Config from target directory
WARNING Making control-plane schedulable by setting MastersSchedulable to true for Scheduler cluster settings
gtema@localhost:test$ grep "99-" .openshift_install_state.json
                "Filename": "openshift/99_openshift-machineconfig_99-master-ssh.yaml",
                "Filename": "openshift/99_openshift-machineconfig_99-worker-ssh.yaml",
gtema@localhost:test$ openshift-install create ignition-configs
INFO Consuming Openshift Manifests from target directory
INFO Consuming Worker Machines from target directory
INFO Consuming Master Machines from target directory
INFO Consuming Common Manifests from target directory
gtema@localhost:test$ grep "99-" .openshift_install_state.json
                        "path": "/opt/openshift/openshift/99_openshift-machineconfig_99-master-ssh.yaml",
                        "path": "/opt/openshift/openshift/99_openshift-machineconfig_99-worker-ssh.yaml",
                        "path": "/opt/openshift/openshift/99_openshift-machineconfig_99-master-ssh.yaml",
                        "path": "/opt/openshift/openshift/99_openshift-machineconfig_99-worker-ssh.yaml",
                "Filename": "openshift/99_openshift-machineconfig_99-master-ssh.yaml",
                "Filename": "openshift/99_openshift-machineconfig_99-worker-ssh.yaml",
                "Filename": "openshift/99_openshift-machineconfig_99-master-ssh.yaml",
                "Filename": "openshift/99_openshift-machineconfig_99-worker-ssh.yaml",
```

So the files are duplicated already in the state file

# What you expected to happen?

those 2 files did not duplicate

# How to reproduce it (as minimally and precisely as possible)?

install_config.yaml:
```
apiVersion: v1
baseDomain: fake.com
compute:
- name: worker
  replicas: 0
controlPlane:
  name: master
  replicas: 3
metadata:
  name: test
networking:
  clusterNetworks:
  - cidr: 192.168.190.0/24
    hostPrefix: 24
  networkType: OpenShiftSDN
  serviceNetwork:
  - 172.30.0.0/16
platform:
  none: {}
pullSecret: '{"auths":{"fake":{"auth": "bar"}}}'
sshKey: ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC9LIyvvV6ur6549MLmwxYid2XV4+GZo0aOsvzNLBZRMx/6GtznMTjxmIRWECKi/M5sqKTpK7TsrWt4kDLXbYUDmVZy+y+/7nEJlpsQmmcr3/1/vKAAt4bDVpbZJyUEMoVT1QOK/IQ2yW0LM3afG43IpeyRk0P13kUvNtrK3dmLBAa6+0OFk0pYx0yv7V6P28P/zct2Wt4dVXpJ13PtxVRUuzBirEv3vJplZtDcEJ9aJl6jVsPrw2VdVEC57hkuFoK+TwKyKCqJheL5mTUnRMKkI4IA29lnhAlR96Gsj1mXltzs/Hl681F0BCXKtnOgcIj0o5uVwnwpWN+U08OF2eYj root@local
```

```console
$ openshift-installer create manifests
$ openshift-installer create ignition-configs
```

# Anything else we need to know?

fcos branch

FCOS has following errors:
10.776386] ignition[597]: error at $.storage.files.48, line 1 col 100405: duplicate entry defined
[ 10.783147] ignition[597]: error at $.storage.files.49, line 1 col 101854: duplicate entry defined
[[0;1;31mFAILED[0m] Failed to start [0;1;39mIgnition (fetch)[0m.

moved from https://github.com/openshift/installer/issues/2975

# References

**Comment by vrutkovs** [comment]
Probably that's related to #37 - installer / terraform-provider-ignition is duplicating files instead of replacing those

**Comment by gtema** [comment]
Don't know, if that might explain duplication already in .openshift_install_state.json - might be

**Comment by LorbusChris** [comment]
Could you try if that works with a 4.4 release image?

If it also happens there it could be related to https://github.com/terraform-providers/terraform-provider-ignition/pull/66#issuecomment-579714329

**Comment by gtema** [comment]
yes it does, the same behavior with preview2

**Comment by gtema** [comment]
well there is anyway need for a doc update in the fcos branch, since bootstrap ignition itself need to be modified for 3.0.0 (https://github.com/openshift/installer/blob/fcos/docs/user/openstack/install_upi.md#bootstrap-ignition-shim)

**Comment by vrutkovs** [comment]
Right, can reproduce. Apparently installer has been doing that since the start and Ignition v2 skipped these files. In Ignition spec3 its invalid

>well there is anyway need for a doc update in the fcos branch

This should be a different bug (feel free to send a PR to `fcos` branch)

**Comment by LorbusChris** [comment]
should be fixed now by installer#3061
/close

**Comment by LorbusChris** [comment]
The solution we tried doesn't work properly. Reopening.

**Comment by vrutkovs** [comment]
Just to clarify - the commited fix makes `create manifests && create cluster` pass - but masters don't get a MC with ssh keys, so users can't ssh there

**Comment by vrutkovs** [comment]
Fixed in #3100
